tests/unit/models/LifecycleConfiguration.js

```diff
@@ -666,5 +666,135 @@ describe('LifecycleConfiguration', () => {
+        it('should not return error when only one transition', () => {
+            const params = {
+                rule: {
+                    Transition: [{
+                        Days: ['0'],
+                        StorageClass: ['a'],
+                    }],
+                },
+                days: 0,
+                storageClass: 'a',
+            };
+            const error = lifecycleConfiguration._checkTimeGap(params);
+            assert.strictEqual(error, undefined);
+        });
+
+        it('should not return error when transitions have days greater than ' +
+        '24 hours apart', () => {
+            const params = {
+                rule: {
+                    Transition: [{
+                        Days: ['0'],
+                        StorageClass: ['a'],
+                    }, {
+                        Days: ['1'],
+                        StorageClass: ['b'],
+                    }],
+                },
+                days: 0,
+                storageClass: 'a',
+            };
+            const error = lifecycleConfiguration._checkTimeGap(params);
+            assert.strictEqual(error, undefined);
+        });
+
+        it('should return error when transitions have same day', () => {
+            const params = {
+                rule: {
+                    Prefix: 'prefix',
+                    Transition: [{
+                        Days: ['0'],
+                        StorageClass: ['a'],
+                    }, {
+                        Days: ['0'],
+                        StorageClass: ['b'],
+                    }],
+                },
+                days: 0,
+                storageClass: 'a',
+            };
+            const error = lifecycleConfiguration._checkTimeGap(params);
+            assert(error.InvalidArgument);
+        });
+
+        it('should not return error when transitions have dates greater than ' +
+        '24 hours apart', () => {
+            const params = {
+                rule: {
+                    Transition: [{
+                        Date: ['2019-01-01T00:00:00.000Z'],
+                        StorageClass: ['a'],
+                    }, {
+                        Date: ['2019-01-02T00:00:00.000Z'],
+                        StorageClass: ['b'],
+                    }],
+                },
+                date: '2019-01-01T00:00:00.000Z',
+                storageClass: 'a',
+            };
+            const error = lifecycleConfiguration._checkTimeGap(params);
+            assert.strictEqual(error, undefined);
+        });
+
+        it('should return error when transitions have dates less than 24 ' +
+        'hours apart', () => {
+            const params = {
+                rule: {
+                    Prefix: 'prefix',
+                    Transition: [{
+                        Date: ['2019-01-01T00:00:00.000Z'],
+                        StorageClass: ['a'],
+                    }, {
+                        Date: ['2019-01-01T23:59:59.999Z'],
```

Comment:
This is *really* on the edge of being valid :wink:

Reply:
😆 I had to draw the line somewhere.